test/system/src/main/java/io/pravega/test/system/framework/services/docker/PravegaControllerDockerService.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system.framework.services.docker;
+
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ServiceCreateResponse;
+import com.spotify.docker.client.messages.mount.Mount;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+import com.spotify.docker.client.messages.swarm.ContainerSpec;
+import com.spotify.docker.client.messages.swarm.EndpointSpec;
+import com.spotify.docker.client.messages.swarm.NetworkAttachmentConfig;
+import com.spotify.docker.client.messages.swarm.PortConfig;
+import com.spotify.docker.client.messages.swarm.ResourceRequirements;
+import com.spotify.docker.client.messages.swarm.Resources;
+import com.spotify.docker.client.messages.swarm.ServiceMode;
+import com.spotify.docker.client.messages.swarm.ServiceSpec;
+import com.spotify.docker.client.messages.swarm.TaskSpec;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+@Slf4j
+public class PravegaControllerDockerService extends DockerBasedService {
+
+    static final int CONTROLLER_PORT = 9092;
+    private static final int REST_PORT = 10080;
+    private int instances = 1;
+    private double cpu = 0.1;
+    private double mem = 700.0;
+
+    public PravegaControllerDockerService(final String serviceName) {
+        super(serviceName);
+    }
+
+    @Override
+    public void stop() {
```

Comment:
This is overriding a method without calling super. Perhaps the parent class could be refactored to make it harder to make this mistake. Perhaps by having an abstract doStop or something. Also look into Guava's Service abstraction. It might simplify things.

Reply:
done.